automation_tools/satellite6/upgrade/__init__.py

```diff
@@ -167,17 +169,20 @@ def product_upgrade(product):
     if check_necessary_env_variables_for_upgrade(product):
         sat_host, cap_hosts, clients6, clients7 = setup_products_for_upgrade(
             product, os.environ.get('OS'))
-        execute(satellite6_upgrade, host=sat_host)
-        # Generate foreman debug on satellite after upgrade
-        execute(foreman_debug, 'satellite_{}'.format(sat_host), host=sat_host)
-        if product == 'capsule':
-            for cap_host in cap_hosts:
-                execute(satellite6_capsule_upgrade, cap_host, host=cap_host)
-                # Generate foreman debug on capsule after upgrade
-                execute(
-                    foreman_debug,
-                    'capsule_{}'.format(cap_host),
-                    host=cap_host)
-        if product == 'client':
-            satellite6_client_upgrade('rhel6', clients6)
-            satellite6_client_upgrade('rhel7', clients7)
+        with LogAnalyzer(sat_host):
```

Comment:
Repeat the LogAnalyzer context manager on sat_host(which you are doing already) and on cap_host.
I will take care about clients as they are situated in Docker and the whole process is bit complex.